File: Data Structures & Algorithms/is-anagram/submission-3.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length())  // first check to avoid creating map unneccesarily 
            return false;

        unordered_map<char, int> countS; // map for first string
        unordered_map<char, int> countT;  // map for second string
        for (int i = 0; i < s.length(); i++) { // add each string element to the map
            countS[s[i]]++;     // map[element] ++ to increment counts of the element
            countT[t[i]]++;
        }
        return countS == countT;
        
        
    }
};

File: Data Structures & Algorithms/is-subsequence/submission-3.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        tp=0 
        sp=0
        while sp<len(s):
            char = s[sp] 
               
            while tp<len(t) and t[tp] != char:
                tp+=1 
        
            if tp >= len(t) or t[tp]!=char:
                return False
            sp+=1
            tp+=1
        return True
        